ccdproc/tests/test_cosmicray.py

```diff
@@ -21,17 +26,22 @@
 
 def add_cosmicrays(data, scale, threshold, ncrays=NCRAYS):
     size = data.shape[0]
-    rng = np.random.default_rng(99)
-    crrays = rng.integers(0, size, size=(ncrays, 2))
+    rng = default_rng(99)
+    crrays = xp.asarray(rng.integers(0, size, size=(ncrays, 2)))
     # use (threshold + 15) below to make sure cosmic ray is well above the
     # threshold no matter what the random number generator returns
     # add_cosmicrays is highly sensitive to the seed
     # ideally threshold should be set so it is not sensitive to seed, but
     # this is not working right now
-    crflux = 10 * scale * rng.random(NCRAYS) + (threshold + 15) * scale
+    crflux = xp.asarray(10 * scale * rng.random(ncrays) + (threshold + 15) * scale)
+
+    # Some array libraries (Dask) do not support setting individual elements,
+    # so use NumPy.
+    data_as_np = np_array(data.data)
     for i in range(ncrays):
         y, x = crrays[i]
-        data.data[y, x] = crflux[i]
+        data_as_np[y, x] = crflux[i]
+    data.data = xp.asarray(data_as_np)
```

Comment:
Does this work for vectorized assignment?

```python
data_as_np = np_array(data.data)
y, x = crrays.T
data_as_np[y, x] = crflux
data.data = xp.asarray(data_as_np)
```

Reply:
That raised an error...given that this is in a test I'm just going to leave it as a loop.